src/shared/openssl-util.c

```diff
@@ -1298,6 +1298,105 @@ int pkey_generate_volume_keys(
                 return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Unsupported public key type: %s", OBJ_nid2sn(type));
         }
 }
+
+static int load_key_from_provider(
+                const char *provider,
+                const char *private_key_uri,
+                EVP_PKEY **ret_private_key) {
+
+        assert(provider);
+        assert(private_key_uri);
+        assert(ret_private_key);
+
+#if OPENSSL_VERSION_MAJOR >= 3
+        if (!OSSL_PROVIDER_try_load(/* ctx= */ NULL, provider, /* retain_fallbacks= */ true))
+                return log_openssl_errors("Failed to load OpenSSL provider '%s'", provider);
+        if (!OSSL_PROVIDER_try_load(/* ctx= */ NULL, "default", /* retain_fallbacks= */ true))
+                return log_openssl_errors("Failed to load OpenSSL provider 'default'");
```

Comment:
That would break the hermetic-usr specification - things should work in a sensible way by default with an empty /etc. If systemd failed to boot unless /etc/systemd/system.conf was present, it would be pretty lousy. There's no reason to fail here, it's trivial to make it work by default.

Reply:
Can we add a comment here explaining the reasoning for doing this? Also, why also load the "default" provider?

Reply:
Loading the default as well seemed to be the suggestion from what I gathered when adding this. It doesn't hurt to do so.
I really do not have the energy to go fight with openssl, but I'll add a comment